docs/resources/ddm_instance.md

```diff
@@ -0,0 +1,174 @@
+
+## Example Usage
+
+```hcl
+resource "huaweicloud_vpc" "test" {
+  name = "test_vpc"
+  cidr = "192.168.0.0/24"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name       = "test_subnet"
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.1"
+  vpc_id     = huaweicloud_vpc.test.id
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name = "test_secgroup"
+}
+
+data "huaweicloud_availability_zones" "test" {}
+
+data "huaweicloud_ddm_engines" test {
+  version = "3.0.8.5"
+}
+
+data "huaweicloud_ddm_flavors" test {
+  engine_id = data.huaweicloud_ddm_engines.test.engines[0].id
+  cpu_arch  = "X86"
+}
+
+resource "huaweicloud_ddm_instance" "test" {
+  name              = "ddm_test"
+  flavor_id         = data.huaweicloud_ddm_flavors.test.flavors[0].id
+  node_num          = 2
+  engine_id         = data.huaweicloud_ddm_engines.test.engines[0].id
+  vpc_id            = huaweicloud_vpc.test.id
+  subnet_id         = huaweicloud_vpc_subnet.test.id
+  security_group_id = huaweicloud_networking_secgroup.test.id
+  
+  availability_zones = [
+    data.huaweicloud_availability_zones.test.names[0]
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the DDM instance.
+  An instance name starts with a letter, consists of 4 to 64 characters, and can contain only letters,
+  digits, and hyphens (-).
+
+* `flavor_id` - (Required, String) Specifies the ID of a product.
+
+* `node_num` - (Required, Int) Specifies the number of nodes.
+
+* `engine_id` - (Required, String, ForceNew) Specifies the ID of an Engine.
+
+  Changing this parameter will create a new resource.
+
+* `availability_zones` - (Required, List, ForceNew) Specifies the list of availability zones.
+
+  Changing this parameter will create a new resource.
+
+* `vpc_id` - (Required, String, ForceNew) Specifies the ID of a VPC.
+
+  Changing this parameter will create a new resource.
+
+* `security_group_id` - (Required, String) Specifies the ID of a security group.
+
+* `subnet_id` - (Required, String, ForceNew) Specifies the ID of a subnet.
+
+  Changing this parameter will create a new resource.
+
+* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project id.
+  Value 0 indicates the default enterprise project.
+
+  Changing this parameter will create a new resource.
+
+* `param_group_id` - (Optional, String, ForceNew) Specifies the ID of parameter group.
+
+  Changing this parameter will create a new resource.
+
+* `time_zone` - (Optional, String, ForceNew) Specifies the time zone.
+
+  Changing this parameter will create a new resource.
+
+* `admin_user_name` - (Optional, String, ForceNew) Specifies the username of the administrator.
```

Comment:
how about use `admin_user` and `admin_password`?

Reply:
it's a good advice,done